sycl/doc/extensions/supported/sycl_ext_oneapi_dot_accumulate.asciidoc

```diff
@@ -95,56 +120,59 @@ int32_t dot_acc(vec<uint8_t,4> a,
+                          int32_t c)
+int32_t doc_acc_4x8packed(uint32_t a,
+                          int32_t b,
+                          int32_t c)
+int32_t doc_acc_4x8packed(uint32_t a,
+                          uint32_t b,
+                          int32_t c);
+----
+
+|Performs a four-component integer dot product accumulate operation, where
+`a` and `b` are 32-bit integers that represent a vector of 4 8-bit elements.
+When the type of `a` or `b` is `int32_t`, it is interpreted as `vec<int8_t,4>`.
+When the type of `a` or `b` is `uint32_t`, it is interpreted as
+`vec<uint8_t,4>`.  In each case, the least significant byte is element 0, and
+the most significant byte is element 3.
 
 |====
 
-== Sample Header
+=== Deprecated functions
+
+The following functions are deprecated.  They have the same effect as the
+`doc_acc_4x8packed` overloads described above.
 
 [source,c++]
 ----
-namespace cl {
-namespace sycl {
-namespace ext {
-namespace oneapi {
-
-int32_t dot_acc(vec<int8_t,4>  a, vec<int8_t,4>  b, int32_t c);
-int32_t dot_acc(vec<int8_t,4>  a, vec<uint8_t,4> b, int32_t c);
-int32_t dot_acc(vec<uint8_t,4> a, vec<int8_t,4>  b, int32_t c);
-int32_t dot_acc(vec<uint8_t,4> a, vec<uint8_t,4> b, int32_t c);
+namespace sycl::ext::oneapi {
 
 int32_t dot_acc(int32_t a, int32_t b, int32_t c);
 int32_t dot_acc(int32_t a, uint32_t b, int32_t c);
 int32_t dot_acc(uint32_t a, int32_t b, int32_t c);
 int32_t dot_acc(uint32_t a, uint32_t b, int32_t c);
 
-} // oneapi
-} // ext
-} // sycl
-} // cl
+} // namespace sycl::ext::oneapi
 ----
 
+
 == Issues
 
-None.
-
-== Revision History
-
-[cols="5,15,15,70"]
-[grid="rows"]
-[options="header"]
-|========================================
-|Rev|Date|Author|Changes
-|1|2019-12-13|Ben Ashbaugh|*Initial draft*
-|2|2019-12-18|Ben Ashbaugh|Switched to standard C++ fixed width types.
-|3|2020-10-26|Rajiv Deodhar|Added int32 types.
-|========================================
-
-//************************************************************************
-//Other formatting suggestions:
-//
-//* Use *bold* text for host APIs, or [source] syntax highlighting.
-//* Use `mono` text for device APIs, or [source] syntax highlighting.
-//* Use `mono` text for extension names, types, or enum values.
-//* Use _italics_ for parameters.
-//************************************************************************
+* The overloads that take two unsigned vectors have a signed `c` and return a
+  signed result.  This is inconsistent with the SPIR-V primitives and the
+  OpenCL C APIs, both of which return an unsigned value in this case and expect
+  an unsigned `c`.  I think we could implement the APIs as they are using the
+  SPIR-V primitives, but the extra unsigned-to-signed conversions might
+  generate less efficient code (I haven't checked).  Is there a compelling
+  reason to keep these APIs as they are now?  If not, we could deprecate them
+  and introduce overloads that take an  unsigned `c` and return an unsigned
+  value.
```

Comment:
Attention reviewer: I'd like to resolve this open issue before merging this PR.  I'm inclined to deprecated the "unsigned / unsigned" overloads as described above in order to align with the SPIR-V extension and the OpenCL C extension.  Does anyone have opinions on why this might be a bad idea?

Tagging @rdeodhar because I think you made the original PR (almost 3 years ago) adding this extension.  Do you remember any rationale about this?

Reply:
I don't recall why the output was always signed. There is no compelling reason to maintain the old API. Consistency with SPIR-V and OpenCL would be peferred.

Reply:
+1, if it's not too late to change to an unsigned accumulator in this case we should try to be consistent.

Although, this should probably "just work" regardless thanks to twos complement magic, at least until we add saturating functions...

Reply:
Changed to an unsigned accumulator and return value in 84f1ff887f8c5cc725ea3248330dd52b5844ea82.  The old version is still available as a deprecated overload.